### PR TITLE
[MI-3560]: Fixed the JS error while posting meeting link to thread issue #300

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
- Fixed issue 300 i.e. Zoom plugin giving JS error while posting meeting link to a thread using the plugin button in the app bar

### Screenshot
Screenshot from 2023-09-26 14-33-06

### CODE REVIEW

File: webapp/package-lock.json
Comment (raghavaggarwal2308) [inline webapp/package-lock.json:1]: Revert these changes

File: webapp/src/index.js [hunk 1]
```diff
@@ -27,8 +27,16 @@ class Plugin {
...
             registry.registerAppBarComponent(
                 iconURL,
-                (channel) => {
-                    startMeeting(channel.id)(store.dispatch, store.getState);
+                async (channel) => {
+                    if (channel) {
+                        startMeeting(channel.id, '')(store.dispatch, store.getState);
+                    } else {
+                        const teamId = store.getState().entities.teams.currentTeamId;
+                        const threadId = store.getState().views.threads.selectedThreadIdInTeam[teamId];
```
Comment (raghavaggarwal2308) [inline webapp/src/index.js:35]: What will be the value of threadId, if we are accessing it inside a channel?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/index.js (hunk 1 above)
Comment (Kshitij-Katiyar) [reply webapp/src/index.js:35]: If there is a thread open in RHS, it gives the threadId of that thread and if no thread is present in rhs, it gives an empty string. Meeting link gets posted to the main channel in case of threadId being empty

Comment (mickmister) [reply webapp/src/index.js:35]: Note that we don't want to change the behavior when we are in center channel view. When a thread is open in the RHS, we should still post in the center channel. The scope of this change should only effect the "Threads View"

Comment (abbas-mehdi-naqvi) [reply webapp/src/index.js:35]: @mickmister The scope of this change only affects the "Threads View", there is no change in the behavior when we are in the center channel view.

Comment (Kshitij-Katiyar) [reply webapp/src/index.js:35]: @mickmister The scope of this change only affects the "Threads view". There is no change in the behavior when we are in the center channel view.

Comment (Kshitij-Katiyar) [reply webapp/src/index.js:35]: @mickmister The scope of this change only affects the "Threads view". There is no change in the behavior when we are in the center channel view.

**Review by raghavaggarwal2308 — APPROVED** [review]
